src/ray/gcs/gcs_server/gcs_placement_group_scheduler.cc

```diff
@@ -480,22 +485,26 @@ void GcsPlacementGroupScheduler::OnAllBundleCommitRequestReturned(
   committed_bundle_location_index_.AddBundleLocations(placement_group_id,
                                                       prepared_bundle_locations);
 
+  // If the placement group scheduling has been cancelled, destroy them.
+  if (lease_status_tracker->GetLeasingState() == LeasingState::CANCELLED) {
```

Comment:
This is another issue I found while I added unit tests. If the lease is cancelled, we should just remove placement groups. Previously, it removes only when `!lease_status_tracker->AllCommitRequestsSuccessful()`